system/core/pages.php

```diff
@@ -51,17 +83,34 @@ protected function replacePageListNode($node) {
 			$this->pagename = $language->getAttribute('name');
 		}
 
-		/**
-		 * Return the given url argument, or null if it is not
-		 * present.
+		/*
+		 * Returns the title to be used for e.g. linking to this
+		 * page.
 		 */
-		protected function getArg($index) {
-			if (array_key_exists($index, $this->args))
-				return $this->args[$index];
-			return null;
+		public function getTitle() {
+			// TODO: Implement a meaningful title in all
+			// subclasses and make this function abstract?
+			return showPagename($this->pagename);
 		}
 
-		abstract function process(HyphaRequest $request);
+		public function renderSingleLine($container) {
+			$h2 = $container->document()->createElement('h2');
```

Comment:
Not sure how DOM is handled but I've learned the best practice is to first create the element and then add it to the DOM (as a non blocking review suggestion :-) )

Reply:
I suspect that in a browser, this has performance advantages (not changing something that is already being rendered), but I also suspect it does not matter much in this case, since the DOM is only kept in memory. Still, it is probably good to stay consistent, so I've changed this.